File: framework/languages/java/samples/java/ShoppingMallCheckout/Server/CommerceApi/src/main/java/systems/zlink/samples/shoppingmallcheckout/server/commerceapi/handlers/DeleteProjectionHandler.java
package systems.zlink.samples.shoppingmallcheckout.server.commerceapi.handlers;

import systems.zlink.framework.channels.ZLinkRequestContext;
import systems.zlink.framework.channels.ZLinkRequestHandler;
import systems.zlink.framework.handlers.ZLinkHandlerGroup;
import systems.zlink.samples.shoppingmallcheckout.server.configuration.CommerceStore;
import systems.zlink.samples.shoppingmallcheckout.shared.contracts.Messages;

@ZLinkHandlerGroup("commerce")
public final class DeleteProjectionHandler
    implements ZLinkRequestHandler<Messages.DeleteProjectionReq, Messages.DeleteProjectionRes> {
    private final CommerceStore store;

    public DeleteProjectionHandler(CommerceStore store) {
        this.store = store;
    }

    @Override
    public Messages.DeleteProjectionRes handle(
        Messages.DeleteProjectionReq request,
        ZLinkRequestContext context) {
        return new Messages.DeleteProjectionRes(store.deleteReadModel(request.orderId()));
    }
}

File: framework/languages/java/samples/kotlin/ShoppingMallCheckout/Server/CommerceApi/src/main/kotlin/systems/zlink/samples/kotlin/shoppingmallcheckout/server/commerceapi/handlers/DeleteProjectionHandler.kt
package systems.zlink.samples.kotlin.shoppingmallcheckout.server.commerceapi.handlers

import systems.zlink.framework.channels.ZLinkRequestContext
import systems.zlink.framework.channels.ZLinkRequestHandler
import systems.zlink.framework.handlers.ZLinkHandlerGroup
import systems.zlink.samples.kotlin.shoppingmallcheckout.server.configuration.CommerceStore
import systems.zlink.samples.kotlin.shoppingmallcheckout.shared.contracts.DeleteProjectionReq
import systems.zlink.samples.kotlin.shoppingmallcheckout.shared.contracts.DeleteProjectionRes

@ZLinkHandlerGroup("commerce")
class DeleteProjectionHandler(
    private val store: CommerceStore,
) : ZLinkRequestHandler<DeleteProjectionReq, DeleteProjectionRes> {
    override fun handle(
        request: DeleteProjectionReq,
        context: ZLinkRequestContext,
    ): DeleteProjectionRes = DeleteProjectionRes(store.deleteReadModel(request.orderId))
}
